src/components/screen/SearchUser.js

```diff
@@ -102,107 +103,101 @@ const StyledSearchImage = styled.Image`
 `;
 
 type Props = {
-
+  navigation: NavigationScreenProps,
 };
 type State = {
   users: Friend[],
   searchedUsers: Friend[],
 };
 
-class Screen extends Component<Props, State> {
-  static navigationOptions = {
-    title: 'Search User',
-  };
-
-  scrollY = new Animated.Value(0);
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      searchedUsers: userSampleData,
-      users: userSampleData,
-    };
-  }
+function Screen(props: Props) {
+  const profileModal = useContext(ProfileModalContext);
+  const [searchedUsers, setSearchedUsers] = useState(userSampleData);
+  const [users, setUsers] = useState(userSampleData);
+  const scrollY = new Animated.Value(0);
 
-  renderItem = (item: Friend, data: any) => {
+  const userListOnPress = (item: Friend) => {
+    if (profileModal && profileModal.dispatch) {
+      profileModal.dispatch({
+        type: 'show-modal',
+        payload: { user: item, deleteMode: true },
+      });
+    }
+  };
+  const renderItem = ({ item, index }: { item: Friend, index: number}) => {
```

Comment:
Why not just
```
const renderItem = ({ item: Friend, index: numnber }) => {
```

?

Reply:
what you've suggested is not working.
when I did this, It throws error. error message is like `'index' is not defined` | `'item' is not defined` in eslint.
This is `destructuring` so I think defining as Flowtype in destructuring is not working.
plz refer to this link -> https://github.com/facebook/flow/issues/235#issuecomment-72592788

Reply:
Ok this looks new to me. Thank you for detail!